cloud/scope/cluster.go

```diff
@@ -57,7 +59,18 @@ func NewClusterScope(params ClusterScopeParams) (*ClusterScope, error) {
 		params.Logger = klogr.New()
 	}
 
-	err := params.AzureClients.setCredentials(params.AzureCluster.Spec.SubscriptionID)
+	var subID string
+	if params.AzureCluster.Spec.SubscriptionID != "" {
+		subID = params.AzureCluster.Spec.SubscriptionID
+	} else {
+		subID = os.Getenv(auth.SubscriptionID)
+	}
+
+	if subID == "" {
+		return nil, fmt.Errorf("error creating azure services. subscriptionID is not set in cluster or AZURE_SUBSCRIPTION_ID env var")
+	}
+
+	err := params.AzureClients.setCredentials(subID)
```

Comment:
not sure I 100% like this better than before, to me it makes sense to do `params.AzureClients.setCredentials(params.AzureCluster.Spec.SubscriptionID)` and override the env value with that string if it's not empty in the setCredentials function 🤔

Reply:
yeah i'll switch this back